### PR TITLE
Add wallet selector and ledger support

### PR DESCRIPTION
**Pull request by bartosz-lipinski**
Added wallet selector and support for three extra wallets:
* Ledger
* Solong (Chrome extension)
* Phantom (Chrome extension)

image

### CODE REVIEW

File: src/wallet-adapters/phantom/index.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      return;
+    }
+
+    if ((window as any)?.solana?.isPhantom) {
+      this._provider = (window as any).solana;
+    } else {
+      window.open('https://phantom.app/', '_blank');
+      notify({
+        message: 'Phantom Error',
+        description: 'Please install Phantom wallet from Chrome ',
```
Comment (fragosti) [inline src/wallet-adapters/phantom/index.tsx:75]: We do plan on supporting multiple platforms, so wouldn't specify chrome.

File: src/wallet-adapters/phantom/index.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      });
+      return;
+    }
+
+    return this._provider?.connect().then(() => this.emit('connect'));
```
Comment (fragosti) [inline src/wallet-adapters/phantom/index.tsx:80]: Why not set this up in the constructor by listening to the connect even on the provider?

File: src/wallet-adapters/phantom/index.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  disconnect() {
+    if (this._provider) {
+      this._provider.disconnect();
+      this._provider = undefined;
+      this.emit('disconnect');
```
Comment (fragosti) [inline src/wallet-adapters/phantom/index.tsx:87]: Same here. It is technically possible for the extension to send the disconnect event if it wants.

**Comment by bartosz-lipinski** [comment]
@fragosti thanks for the review. @nathanielparke when you have a moment it would be awesome to integrate this change.

**Review by armaniferrante — APPROVED** [review]

**Pull request merged**